### ISSUE TITLE
Revisit Data Table Design (ORM classes)

### ISSUE THREAD

**Issue by runqingz** [runqingz/MyWallet#14]
<del>To not over engineer data relationships following changes will be implemented:<del>

<del>1. Project should be rework into a task grouping entity (e.g. my family) to enable user Group Management (start/end dates). This should only be available to users with privilege to manage groups. Normal users should only be able to managing tasks.<del>
<del>2. Task should have types (defined enum as cardinal) and tags (defined enum as cardinal) in data table). This should allow for statistical queries and analyzing data (e.g. type line plot, my top tags, worst tags)<del>
<del>3. Task should have a score to evaluate performance (count) against goal (Gain/Loss)<del>

   <del> -  As a starting point: Gain = performance / goal, loss = 1 -  (performance - goal) / goal<del>
